### ISSUE TITLE
Memory error in footprint for sparce matrix 

### ISSUE THREAD

**Issue by dremovd** [vecxoz/vecstack#29]
X is
`<239761x68891 sparse matrix of type '<class 'numpy.float64'>'
	with 8726453 stored elements in Compressed Sparse Row format>`

Specifically, choice function is crashing, because n==16517375051
Error is:
```MemoryError                               Traceback (most recent call last)
~/.local/lib/python3.5/site-packages/vecstack/coresk.py in _get_footprint(self, X, n_items)
    863             # np.random.seed(0) # for development
--> 864             ids = np.random.choice(n, min(n_items, n), replace=False)
    865 

mtrand.pyx in mtrand.RandomState.choice()

mtrand.pyx in mtrand.RandomState.permutation()

MemoryError: 

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-24-4be4f86278e7> in <module>()
      8                             verbose=2)             
      9 t = targets[0]
---> 10 stack = stack.fit(X, y[t])

~/.local/lib/python3.5/site-packages/vecstack/coresk.py in fit(self, X, y, sample_weight)
    393             self.n_classes_ = None
    394         self.n_estimators_ = len(self.estimators_)
--> 395         self.train_footprint_ = self._get_footprint(X)
    396 
    397         # ---------------------------------------------------------------------

~/.local/lib/python3.5/site-packages/vecstack/coresk.py in _get_footprint(self, X, n_items)
    872 
    873         except Exception:
--> 874             raise ValueError('Internal error. '
    875                              'Please save traceback and inform developers.')
    876 

ValueError: Internal error. Please save traceback and inform developers.```

**Comment by vecxoz** [comment]
Thanks for report! Nice catch!

When first argument `n` of `np.random.choice` is an integer numpy does `np.arange(n)` internally i.e. it creates the whole array with shape `(16517375051, )`. So out-of-memory in this case is kind of expected behavior. :) Practically it means that `choice` function actually was not a good choice. We just need `np.random.randint`. I’m working on the fix.

Meanwhile you can try to use `stacking` function. Its output is absolutely the same, but it does not use `choice` inside:

`from vecstack import stacking`

**Comment by dremovd** [comment]
Thanks for this comment, I actually was looking for substitute like stacking

**Comment by vecxoz** [comment]
I fixed #29.
You can reinstall package from github directly (without updating dependencies):

`pip install --upgrade --no-deps https://github.com/vecxoz/vecstack/archive/master.zip`
